src/Event/BaseEvent.php

```diff
@@ -36,15 +36,17 @@ abstract class BaseEvent implements EventInterface
      * @param string $source The name of the source (table)
      * @param array $changed The array of changes that got detected for the entity
      * @param array $original The original values the entity had before it got changed
+     * @param \Cake\Datasource\EntityInterface|null $entity The entity being changed
      */
-    public function __construct($transactionId, $id, $source, $changed, $original)
+    public function __construct($transactionId, $id, $source, $changed, $original, $entity)
```

Comment:
Shoudn't it be optional? So that existing events dont fail?

```suggestion
    public function __construct($transactionId, $id, $source, $changed, $original, $entity = null)
```

Reply:
I think I didn't do this because events get created from the EventFactory. However, this won't hurt and I suppose some folks may have decided to extend internals so this _could be_ a breaking change.